File: src/solution_java/easy/_66PlusOne.java
package solution_java.easy;

public class _66PlusOne {
    class Solution {
        public int[] plusOne(int[] digits) {
            for (int i = digits.length - 1; i >= 0; i--) {
                if (digits[i] == 9) {
                    digits[i] = 0;
                } else {
                    digits[i] += 1;
                    return digits;
                }
            }
            int[] temp = new int[digits.length + 1];
            temp[0] = 1;
            return temp;
        }
    }
}

File: src/easy/_66PlusOne.kt
package easy

class _66PlusOne {
    class Solution {
        fun plusOne(digits: IntArray): IntArray {
            for (i in (digits.size - 1) downTo 0) {
                if (digits[i] == 9) {
                    digits[i] = 0
                } else {
                    digits[i] += 1
                    return digits
                }
            }
            val tempDigits = IntArray(digits.size + 1)
            tempDigits[0] = 1
            return tempDigits
        }
    }
}